TestResultSummaryService/parsers/Parser.js

```diff
@@ -4,6 +4,21 @@ class Parser {
     }
 
     static canParse() { return false; }
+    exactJavaVersion(output) {
+        const javaVersionRegex = /((openjdk|java) version[\s\S]*JCL.*\n|(openjdk|java) version[\s\S]*Server VM.*\n)/;
+        let curRegexResult = null;
+        if ( ( curRegexResult = javaVersionRegex.exec( output ) ) !== null ) {
+            return curRegexResult[1];
+        }
+    }
+
+    exactJavaBuildDate(output) {
+        let curRegexResult = null;
+        const javaBuildDateRegex = /-(20[0-9][0-9][0-9][0-9][0-9][0-9])/;

```

Comment:
This `exactJavaBuildDate ` will match any date that starts with `-20`. It may not be a date from java version output.
`exactJavaVersion` gets the java output already. We can further process its output and get java build date. We need only one function.

Reply:
Removed the extractJavaBuildDate function as required.